packages/unified-selection/src/unified-selection/SelectionStorage.ts

```diff
@@ -83,6 +86,16 @@ export interface SelectionStorage {
     /** Key of the iModel to change selection for. */
     iModelKey: string;
   }): void;
+
+  /** Get the current hilite set for the specified imodel */
+  getHiliteSet(props: {
```

Comment:
I'm hesitant on whether we want to have `getHiliteSet` on a `SelectionStorage` interface... I know it's convenient, but maybe we can come up with a way to keep the benefit of "one hilite set request per selection" without `SelectionStorage` knowing about hilite sets?

Reply:
> Could you elaborate on the "one hilite set request per selection" part? 

The current implementation in `@itwin/presentation-frontend`, as well as yours here, cache hilite set per iModel. In addition, `HiliteSetProvider` in `@itwin/presentation-frontend` caches the result based on input keys - requesting hilite set for the same keys on the same `HiliteSetProvider` doesn't require re-running the requests/queries. This is an important piece of functionality when retrieving hilite sets for selection, because a hilite set may be requested multiple times by multiple different components on a selection change, for the same selected items.

On the other hand, I'd like selection storage to not know anything about hilite sets. Ideally, hilite sets should also not know anything about selection storage and just know about `Selectable`. Maybe we need another concept that takes care of caching hilite sets and invalidating them on selection change?..